File: Source/DATASourceCollectionHeaderView.m
#import "DATASourceCollectionHeaderView.h"

NSString * const DATASourceCollectionHeaderViewIdentifier = @"DATASourceCollectionHeaderViewIdentifier";

@interface DATASourceCollectionHeaderView()

@property (nonatomic) UILabel *titleLabel;

@end

@implementation DATASourceCollectionHeaderView

#pragma mark - Initializers

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];

    if (self) {
        [self addSubview:self.titleLabel];
    }

    return self;
}

#pragma mark - Getters

- (UILabel *)titleLabel {
    if (!_titleLabel) {
        CGFloat margin = 0.0;
        CGRect frame = CGRectMake(margin, 0.0, self.frame.size.width - (margin * 2.0), self.frame.size.height);
        _titleLabel = [[UILabel alloc] initWithFrame:frame];
        _titleLabel.font = [UIFont systemFontOfSize:18.0];
    }

    return _titleLabel;
}

#pragma mark - Public methods

- (void)updateTitle:(NSString *)title {
    self.titleLabel.text = title;
}

@end

File: Source/DATASourceCollectionHeaderView.swift
import UIKit

public class DATASourceCollectionViewHeader: UICollectionReusableView {

    // MARK: Variables

    public static let Identifier = "DATASourceCollectionHeaderViewIdentifier"

    public var title: String = "" {
        didSet {
            self.titleLabel.text = title
        }
    }

    private lazy var titleLabel: UILabel = {
        let label = UILabel(frame: CGRect(x: 0, y: 0, width: self.frame.width, height: self.frame.height))
        label.font = UIFont.boldSystemFontOfSize(22.0)

        return label
    }()

    private lazy var separatorView: UIView = {
        let bottomMargin = CGFloat(10)
        let view = UIView(frame: CGRect(x: 0, y: self.frame.height - bottomMargin, width: self.frame.width, height: 1))
        view.backgroundColor = UIColor.blackColor()

        return view
    }()

    // MARK: Initializers

    public override init(frame: CGRect) {
        super.init(frame: frame)

        self.addSubview(self.titleLabel)
        self.addSubview(self.separatorView)
    }

    required public init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
